presto-main/src/main/java/com/facebook/presto/operator/WindowOperator.java

```diff
@@ -474,51 +474,23 @@ private static int findGroupEnd(PagesIndex pagesIndex, PagesHashStrategy pagesHa
     static int findEndPosition(int startPosition, int endPosition, BiPredicate<Integer, Integer> comparator)
     {
         checkArgument(startPosition >= 0, "startPosition must be greater or equal than zero: %s", startPosition);
-        checkArgument(endPosition > 0, "endPosition must be greater than zero: %s", endPosition);
-        checkArgument(startPosition < endPosition, "startPosition must be less than endPosition: %s < %s", startPosition, endPosition);
+        checkArgument(startPosition < endPosition, "startPosition (%s) must be less than endPosition (%s)", startPosition, endPosition);
 
         int left = startPosition;
-        int right = endPosition - 1;
-        for (int i = 0; i < endPosition - startPosition; i++) {
-            int distance = right - left;
+        int right = endPosition;
 
-            if (distance == 0) {
-                return right + 1;
-            }
+        while (left + 1 < right) {
+            int middle = (left + right) >>> 1;
```

Comment:
I think that: `int mid = left + distance / 2` was more readable. I double there is performance difference